src/fides/api/ops/api/v1/endpoints/storage_endpoints.py

```diff
@@ -278,12 +290,212 @@ def delete_config_by_key(
     """
     logger.info("Finding storage config with key '{}'", config_key)
 
-    storage_config = StorageConfig.get_by(db, field="key", value=config_key)
+    storage_config: Optional[StorageConfig] = StorageConfig.get_by(
+        db, field="key", value=config_key
+    )
     if not storage_config:
         raise HTTPException(
             status_code=HTTP_404_NOT_FOUND,
             detail=f"No configuration with key {config_key}.",
         )
+    if storage_config.is_default:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail="Default storage configurations cannot be deleted.",
+        )
 
     logger.info("Deleting storage config with key '{}'", config_key)
     storage_config.delete(db)
+
+
+@router.put(
+    STORAGE_DEFAULT,
+    status_code=HTTP_200_OK,
+    dependencies=[Security(verify_oauth_client, scopes=[STORAGE_CREATE_OR_UPDATE])],
+    response_model=StorageDestinationResponse,
+)
+def put_default_config(
+    *,
+    db: Session = Depends(deps.get_db),
+    incoming_storage_config: StorageDestinationBase,
+) -> StorageConfig:
+    """
+    Create or update the default storage configuration for the given storage type
+    """
+    logger.info(
+        "Starting upsert for default storage of type '{}'", incoming_storage_config.type
+    )
+
+    incoming_data = incoming_storage_config.dict()
+    existing_default = default_storage_config_by_type(db, incoming_storage_config.type)
+    if existing_default:
+        # take the key of the existing default and add that to the incoming data, to ensure we overwrite the same record
+        incoming_data["key"] = existing_default.key
+    else:
+        # set a name for our config if we're creating a new default
+        incoming_data["name"] = default_storage_config_name(
+            incoming_storage_config.type  # type: ignore
+        )
+
+    # since we're setting the default storage config for the given type, `is_default` MUST be set to `True`
+    incoming_data["is_default"] = True
+
+    try:
+        storage_config = StorageConfig.create_or_update(db=db, data=incoming_data)
+        return storage_config
+    except KeyOrNameAlreadyExists as exc:
+        logger.warning(
+            "Create/update failed for default config update for storage type {}: {}",
+            incoming_storage_config.type,
+            exc,
+        )
+        raise HTTPException(status_code=HTTP_400_BAD_REQUEST, detail=exc.args[0])
+    except Exception as exc:
+        logger.warning(
+            "Create/update failed for default config update for storage type {}: {}",
+            incoming_storage_config.type,
+            Pii(str(exc)),
+        )
+        raise HTTPException(
+            status_code=HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Error creating or updating storage config.",
```

Comment:
good call out. i was sort of just following what is done in the analogous endpoint for non-default configs - https://github.com/ethyca/fides/blob/4ef924acf40383378bb144b0583bfd280831e49a/src/fides/api/ops/api/v1/endpoints/storage_endpoints.py#L160

i'm guessing the reasoning was similar to what you said, or maybe perhaps to avoid exposing potentially sensitive data? that being said, i'm not necessarily opposed to giving more back here, i also considered doing it. so if you think it's the better thing to do, i'm happy to go ahead and udpate.

Reply:
Since this is the current convention let's just stick with it. I'm guessing this became the convention to prevent leaking something.